public/src/components/channelManagement/ChannelSwitches.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useEffect, useState } from 'react';
+import { makeStyles } from '@material-ui/core';
+import {
+  fetchFrontendSettings,
+  FrontendSettingsType,
+  saveFrontendSettings,
+} from '../../utils/requests';
+import Switch from '@material-ui/core/Switch';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Button from '@material-ui/core/Button';
+
+const useStyles = makeStyles(() => ({
+  container: {
+    margin: '30px',
+    maxWidth: '500px',
+    display: 'flex',
+    flexDirection: 'column',
+    '& > * + *': {
+      marginTop: '5px',
+    },
+  },
+}));
+
+type SwitchName = 'enableBanners' | 'enableEpics';
+
+type ChannelSwitches = {
+  [key in SwitchName]: boolean;
+};
+
+const defaultSwitches: ChannelSwitches = {
+  enableBanners: false,
+  enableEpics: false,
+};
+
+interface ChannelSwitchProps {
+  name: SwitchName;
+  label: string;
+  enabled: boolean;
+  setSwitch: (name: SwitchName, enabled: boolean) => void;
+}
+
+const ChannelSwitch: React.FC<ChannelSwitchProps> = ({
+  name,
+  label,
+  enabled,
+  setSwitch,
+}: ChannelSwitchProps) => (
+  <FormControlLabel
+    key={name}
+    label={label}
+    control={
+      <Switch
+        checked={enabled}
+        onChange={(event): void => {
+          setSwitch(name, event.target.checked);
+        }}
+        value={name}
+      />
+    }
+  />
+);
+
+const ChannelSwitches: React.FC = () => {
+  const classes = useStyles();
+  const [switches, setSwitches] = useState<ChannelSwitches>(defaultSwitches);
+  const [version, setVersion] = useState<string | null>(null);
```

Comment:
Looks great! This might not actually save too much code, but we do have a [withS3Data hoc](https://github.com/guardian/support-admin-console/blob/master/public/src/hocs/withS3Data.tsx) which I think would fit here. We're currently using it for the [amounts editor](https://github.com/guardian/support-admin-console/blob/master/public/src/components/amounts/configuredAmountsEditor.tsx#L116), but it was designed (🤞) to be generic!

Reply:
ahh I forgot about this!

Reply:
I've pushed that up - much nicer

Reply:
oh nice - yeah that worked out great! 🚀